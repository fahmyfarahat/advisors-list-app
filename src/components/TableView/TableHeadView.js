import React from "react";
import PropTypes from "prop-types";
import {
  TableHead,
  TableRow,
  TableCell,
  TableSortLabel
} from "@material-ui/core";

const headCells = [
  { id: "name", numeric: false,label: "Advisors" },
  { id: "status", numeric: true, label: "Status" },
  { id: "language", numeric: false, label: "Language" },
  { id: "reviews", numeric: false, label: "Reviews" },
];

const TableHeadView = ({ 
  order, 
  orderBy, 
  onRequestSort 
}) => {
  const sortHandler = property => event => {
    onRequestSort(event, property);
  };

  return (
    <TableHead>
      <TableRow>
        {headCells.map(headCell => (
          <TableCell
            key={headCell.id}
            align={headCell.numeric ? "right" : "left"}
            sortDirection={orderBy === headCell.id ? order : false}
          >
            <TableSortLabel
              active={orderBy === headCell.id}
              direction={order}
              onClick={sortHandler(headCell.id)}
            >
              {headCell.label}
            </TableSortLabel>
          </TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
}

TableHeadView.propTypes = {
  onRequestSort: PropTypes.func.isRequired,
  order: PropTypes.oneOf(["asc", "desc"]).isRequired,
  orderBy: PropTypes.string.isRequired
};

export default TableHeadView;