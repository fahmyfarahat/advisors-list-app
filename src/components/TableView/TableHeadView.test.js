import React from "react";
import { shallow } from "enzyme";
import TableHeadView from "./TableHeadView";
import toJson from "enzyme-to-json";

describe("TableHeadView - general ", () => {
  it("should match Snapshot", () => {
    const props = {
      onRequestSort: () => {},
      order: "asc",
      orderBy: ""
    };
    const wrapper = shallow(<TableHeadView {...props} />);
    expect(wrapper).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
