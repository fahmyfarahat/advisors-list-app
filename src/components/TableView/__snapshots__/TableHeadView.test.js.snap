// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeadView - general  should match Snapshot 1`] = `
<WithStyles(ForwardRef(TableHead))>
  <WithStyles(ForwardRef(TableRow))>
    <WithStyles(ForwardRef(TableCell))
      align="left"
      key="name"
      sortDirection={false}
    >
      <WithStyles(ForwardRef(TableSortLabel))
        active={false}
        direction="asc"
        onClick={[Function]}
      >
        Advisors
      </WithStyles(ForwardRef(TableSortLabel))>
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      align="right"
      key="status"
      sortDirection={false}
    >
      <WithStyles(ForwardRef(TableSortLabel))
        active={false}
        direction="asc"
        onClick={[Function]}
      >
        Status
      </WithStyles(ForwardRef(TableSortLabel))>
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      align="left"
      key="language"
      sortDirection={false}
    >
      <WithStyles(ForwardRef(TableSortLabel))
        active={false}
        direction="asc"
        onClick={[Function]}
      >
        Language
      </WithStyles(ForwardRef(TableSortLabel))>
    </WithStyles(ForwardRef(TableCell))>
    <WithStyles(ForwardRef(TableCell))
      align="left"
      key="reviews"
      sortDirection={false}
    >
      <WithStyles(ForwardRef(TableSortLabel))
        active={false}
        direction="asc"
        onClick={[Function]}
      >
        Reviews
      </WithStyles(ForwardRef(TableSortLabel))>
    </WithStyles(ForwardRef(TableCell))>
  </WithStyles(ForwardRef(TableRow))>
</WithStyles(ForwardRef(TableHead))>
`;
